packages/yup/tests/date.test.js

```diff
@@ -4,10 +4,14 @@ const INVALID = 'Date is invalid.';
 
 const defaults = {};
 
-const validate = async (date, { format, min, max, message } = defaults) => {
+const validate = async (
+  date,
+  { format, min, max, message, typeErrorMessage } = defaults
+) => {
   let schema = avDate({
     format,
-  });
+    typeErrorMessage,
+  }).typeError();
```

Comment:
Since the tests fail without this addition I think it would be a breaking change to anyone who upgrades and didn't happen to be using a `.typeError()` test already. Is there some way we can mutate the schema or constructor in `date.js` so that the initial call to `super()` that runs 
```js
this.withMutation(function () {
    _this.typeError(_locale.mixed.notType);
  });
```
 works, and then the custom message is updated afterwards?

Maybe there is something in [transform](https://github.com/jquense/yup#mixedtransformcurrentvalue-any-originalvalue-any--any-schema) or [withMutation](https://github.com/jquense/yup#mixedwithmutationbuilder-current-schema--void-void) that would help us accomplish that